api/proto/teleport/legacy/types/types.proto

```diff
@@ -2096,6 +2096,33 @@ message CertExtension {
   string Value = 4 [(gogoproto.jsontag) = "value"];
 }
 
+// PolicyV1 is a predicate policy used for RBAC, not dissimilar to Role.
+message PolicyV1 {
+  // Kind is a resource kind
+  string Kind = 1 [(gogoproto.jsontag) = "kind"];
+  // SubKind is an optional resource sub kind, used in some resources
+  string SubKind = 2 [(gogoproto.jsontag) = "sub_kind,omitempty"];
+  // Version is version
+  string Version = 3 [(gogoproto.jsontag) = "version"];
+  // Metadata is resource metadata
+  Metadata Metadata = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "metadata"
+  ];
+  // Spec is a policy specification.
+  PolicySpecV1 Spec = 5 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "spec"
+  ];
+}
+
+// PolicySpecV1 is a predicate policy specification used for RBAC, not dissimilar to Role.
+message PolicySpecV1 {
+  map<string, string> Allow = 1 [(gogoproto.jsontag) = "allow,omitempty"];
+  map<string, string> Deny = 2 [(gogoproto.jsontag) = "deny,omitempty"];
```

Comment:
I'm not sure what's going in these fields, comments could be helpful. A map at the top level `allow` might be a bit hard to extend, if you had a struct it could be easy to add new fields in the future.